### ISSUE TITLE
Blank or invalid format date causes picker to return no value

### ISSUE THREAD

**Issue by gcoluni** [YouCanBookMe/react-datetime#560]
[X ] Bug report
[ ] Feature request
[ ] Support request

### Steps to Reproduce
1. Go to the component.
2. If there is no value, select a value.
3. Click back into the component.
4. Delete the value in the component. 
5. Without clicking out of the component, click in the component again so the date picker appears
6. Pick a date
7. Expected - the new picked date appears
   Actual - nothing appears in the component

### Expected Results
The new picked date appears

### Actual Results
The component still displays a blank value

### Minimal Reproduction of the Problem

![date_picker_issue](https://user-images.githubusercontent.com/20824083/41247553-4b8662b6-6d7c-11e8-8b3d-96127656ee5b.gif)

**Comment by arqex** [comment]
I can't reproduce it. For me it's working ok, is there any other library that might be causing some kind of conflict?

![datepicker](https://user-images.githubusercontent.com/6509397/43783334-c3388ad4-9a61-11e8-951d-d0f0bec4cfeb.gif)

**Comment by gcoluni** [comment]
I don't believe so since the events that are associated with the picker are not even being fired. In all other cases they fire except for that one.

**Comment by Fabianopb** [comment]
@gcoluni I couldn't reproduce either. Could you maybe provide a simple repo where we could reproduce and test the issue?

**Comment by gcoluni** [comment]
Let me take a look at this again. I was pretty sure that nothing was being fired in order to update the value. I will also check what version I am on and see if it is the latest and test.
